packages/fluence.lib.litecoin/src/services/transports/bc.litecoin.transport.ts

```
@@ -0,0 +1,192 @@
+import BigNumber from 'bignumber.js';
+import { get } from 'lodash';
+import * as logger from 'winston';
+import { LitecoinBlock, LitecoinTransaction } from '../../types';
+import { Errors } from '../../errors';
+import { LITECOIN } from '../../constants';
+import { LitecoinTransportProvider } from './interfaces';
+
+const Client = require('bitcoin-core');
+
+export class BcLitecoinTransportService extends Service implements LitecoinTransportProvider {
+    private client: any;
+    protected networkId: string;
+    protected transportConnection: Scheme.TransportConnection;
+
+    constructor(pluginManager: PluginManager, transportConnection: Scheme.TransportConnection) {
+        super(pluginManager);
+
+        this.transportConnection = transportConnection;
+        this.networkId = transportConnection.networkId;
+
+        this.networkId = this.transportConnection.networkId;
+        this.client = new Client(this.transportConnection.settings);
+    }
+
+    public getBlockchainId() {
+        return LITECOIN;
+    }
+
+    public getNetworkId() {
+        return this.networkId;
+    }
+
+    public getTransportConnection() {
+        return this.transportConnection;
+    }
+
+    protected prepareHash(hash: string) {
+        return hash.indexOf('0x') === 0 ? hash.slice(2) : hash;
+    }
+
+    public getServiceId() {
+        return `bitcoin.core.litecoin.${ this.networkId }.transport.service`;
+    }
+
+    public getTransportId() {
+        return `bitcoin.core.litecoin.${ this.networkId }.transport.service`;
+    }
+
+    public async getBalance(address: string, minConf: number = 1): Promise<BigNumber> {
+        const preparedHash = this.prepareHash(address);
+
+        try {
+            const balance = await this.client.getBalance(preparedHash, minConf);
+    
+            return new BigNumber(balance);
+        } catch (ex) {
+            logger.error(`Can't get balance of address [${ address }]. reason: ${ ex.message }`);
```

throw exceptions instead of just logging

---

related to litecoin PR:
https://github.com/fluencesh/fluence.libs.monorepo/pull/4